src/OpenTK.Windowing.Desktop/NativeWindow.cs

```diff
@@ -37,7 +37,7 @@ public class NativeWindow : IDisposable
         private Vector2 _lastReportedMousePos;
 
         // Stores exceptions thrown in callbacks so that we can rethrow them after ProcessEvents().
-        private List<ExceptionDispatchInfo> _callbackExceptions = new List<ExceptionDispatchInfo>();
+        private static List<ExceptionDispatchInfo> _callbackExceptions = new List<ExceptionDispatchInfo>();
```

Comment:
I worry about the `static` here.  If there are multiple `NativeWindow` instances, each running on a different thread, not only could their exceptions get mixed up with each other, but also the `List`'s internal state could get corrupted too.  (Even if it was a `ConcurrentQueue` instead to avoid the state corruption, there'd still be the cross-thread mixing issue.)  If the `static` can be removed, it probably should be; or this is likely a breaking change.

Reply:
I think a `ConcurrentQueue` is the way to go here.
There will only ever be one thread calling `glfwPollEvents()`, calling it from two threads at the same time is an error according to the glfw documentation. `This function must only be called from the main thread.` .

Though callbacks can be triggered on other threads by calling other functions, so I think it's a good idea to use a `ConcurrentQueue`.

This feature is mostly so that linux and mac doesn't freak out if we leak an unhandled exception in a callback, so rethrowing the exception on another thread isn't really a problem (but ideally we want to also give information about what thread caused the exception).

Reply:
Okay.  Makes me a little nervous to put the data from many different threads into a single bucket, but if you're convinced that that's not a problem, I can at least accept `ConcurrentQueue` there.  `List` is dangerously unsafe, but at least `ConcurrentQueue` won't break and it will ensure that exceptions get delivered, even if possibly delivered on the wrong thread.

Reply:
I've changed to `ConcurrentQueue`, with it's `TryDequeue` and `Count` I've decided that any exceptions that are counted in `Count` when we read from it are going to be considered for rethrowing. 
`RethrowExceptionsIfNeeded` should only be called from one thread so `TryDequeue` should success for all the calls we are making to it, but I decided to wrap it in an `if` statement even though it should never be needed.